src/main/scala/edu/berkeley/ce/sparkrocks/Block.scala

```diff
@@ -532,41 +571,70 @@ case class Block(center: Array[Double], faces: Seq[Face], generation: Int=0) ext
         allVertices.map { v => v(2) }.sum / allVertices.size
       )
     } else {
-      Array(
-        // Factor of 3 comes from: centroid / (2.0 * (volume/6.0))
-        3.0 * centroidX / totalVolume,
-        3.0 * centroidY / totalVolume,
-        3.0 * centroidZ / totalVolume
-      )
+      val increments = faces.flatMap { face =>
```

Comment:
How about just doing `vertices.flatMap { case (face, faceVertices) => ... }` to simplify things and get rid of the annoying `vertices(face)` usages? Also, can we rename `vertices` to something that better reflects the fact that it's a `Map` and not a `Seq`?